etl/steps/data/garden/health/2023-08-09/unaids.py

```diff
@@ -102,6 +125,39 @@ def run(dest_dir: str) -> None:
     ds_garden.save()
 
 
+def add_regions_to_tuberculosis_vars(tb: Table, ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
+    """
+    Adding regional aggregates for all tuberculosis variables.
+    """
+    tb_cols = ["hiv_tb_patients_receiving_art", "tb_patients_tested_positive_hiv", "tb_related_deaths"]
+
+    tb_agg = tb[["country", "year", "disaggregation"] + tb_cols]
+    tb_no_agg = tb.drop(columns=tb_cols)
+    # removing existing aggregates - just world
+    tb_agg = tb_agg[~tb_agg["country"].isin(REGIONS_TO_ADD)]
+    tb_agg = tb_agg.dropna(subset=tb_cols, how="all")
+
+    tb_group = tb_agg.groupby("disaggregation")
+
+    tb_all_group = Table()
+    for group, group_tb in tb_group:
+        tb_agg_group = group_tb.drop(columns="disaggregation")
+        tb_agg_group = geo.add_regions_to_table(
+            tb_agg_group,
+            ds_regions,
+            ds_income_groups,
+            REGIONS_TO_ADD,
+            min_num_values_per_year=1,
+            frac_allowed_nans_per_year=0.3,
+        )
+        tb_agg_group["disaggregation"] = group
+        tb_all_group = pr.concat([tb_all_group, tb_agg_group], ignore_index=True)
+
+    tb = pr.merge(tb_no_agg, tb_all_group, on=["country", "year", "disaggregation"], how="outer").reset_index(drop=True)
```

Comment:
I may not fully understand this function. What is the purpose of splitting `tb` into `tb_agg` and `tb_no_agg`? To me, it looks like the only difference between `tb` and `tb_all_group` is that the former will just have more missing values in the indicator columns?


What do you think about the following re-styling? Basically, I've added comments and changed variable names to be slightly more informative IMO:

```suggestion
    cols = ["hiv_tb_patients_receiving_art", "tb_patients_tested_positive_hiv", "tb_related_deaths"]
    
    # Split table into 'agg' (?) and 'no_agg' (?)
    tb_agg = tb[["country", "year", "disaggregation"] + cols]
    tb_no_agg = tb.drop(columns=cols)
    # Removing existing aggregates
    tb_agg = tb_agg[~tb_agg["country"].isin(REGIONS_TO_ADD)]
    tb_agg = tb_agg.dropna(subset=cols, how="all")

    # Group tb_agg by disaggregation for some reason
    tb_agg_groups = tb_agg.groupby("disaggregation")
   
    # Create a table for each disaggregation value
    tbs = []
    for group_name, tb_agg_group in tb_agg_groups:
        tb_agg_group_with_regions = geo.add_regions_to_table(
            tb_agg_group.drop(columns="disaggregation"),
            ds_regions,
            ds_income_groups,
            REGIONS_TO_ADD,
            min_num_values_per_year=1,
            frac_allowed_nans_per_year=0.3,
        )
        tb_agg_group_with_regions["disaggregation"] = group_name
        tbs.append(tb_agg_group_with_regions)
    
    # Combine all 'agg' tables
    tb_agg = pr.concat(tbs, ignore_index=True)

    # Merge with 
    tb = pr.merge(tb_no_agg, tb_agg, on=["country", "year", "disaggregation"], how="outer").reset_index(drop=True)
```

Reply:
Thanks Lucas - that makes sense and I've implemented it!